samcli/lib/build/app_builder.py

```diff
@@ -361,7 +380,7 @@ def _build_layer(
                     build_runtime = compatible_runtimes[0]
             options = ApplicationBuilder._get_build_options(layer_name, config.language, None)
 
-            build_method(config, code_dir, artifacts_dir, scratch_dir, manifest_path, build_runtime, options)
+            build_method(config, code_dir, artifacts_dir, scratch_dir, manifest_path, build_runtime, options, env_vars)
```

Comment:
What is gonna happen to `env_vars` when `container_manager` is `None`?

about the naming `env_vars`, it might cause a bit confusion here. I prefer to name it the full `container_env_vars` because `env_vars` in only used when `if self._container_manager:` is true.
Otherwise I was like "why env_vars is not used when we don't use container" if I don't have the full picture.

Reply:
actually we were thinking about removing the word container even, I'll leave this discussion with Mark and UX team and come back to this